torchtune/models/clip/_tokenizer.py

```diff
@@ -0,0 +1,211 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+import gzip
+import html
+from pathlib import Path
+from typing import List
+
+import ftfy
+import regex as re
+import torch
+
+from torchtune.modules.tokenizers._utils import BaseTokenizer
+
+
+class CLIPTokenizer(BaseTokenizer):
+    """
+    Text tokenizer for CLIP.
+
+    Based on the official implementation here:
+    https://github.com/openai/CLIP/blob/main/clip/simple_tokenizer.py
+
+    Args:
+        vocab_path (Path): the path to the CLIP vocab file
+        max_seq_len (int): the context length (all CLIP models use 77)
+        truncate (bool): whether to truncate the text when longer than max_seq_len
+    """
+
+    def __init__(self, vocab_path: Path, max_seq_len: int = 77, truncate: bool = True):
+        self.max_seq_len = max_seq_len
+        self.truncate = truncate
+
+        self.byte_encoder = _bytes_to_unicode()
+        self.byte_decoder = {v: k for k, v in self.byte_encoder.items()}
+
+        with gzip.open(vocab_path) as f:
+            merges = f.read().decode("utf-8").split("\n")
+        merges = [tuple(merge.split()) for merge in merges[1:48895]]
```

Comment:
the CLIP tokenizer awkwardly uses just the first ~1/5th of its vocab file, resulting in this magic number 48895 and making this class more fragile (e.g. if someone tries to pass in their own custom vocab file, it'll get unexpectedly truncated). It might be worth manually truncating the file itself and using that instead (but we'd need to add it to git or host it somewhere). Or alternatively, we could use huggingface's clip vocab, which I assume has already done this